test/test_python_dispatch.py

```diff
@@ -551,6 +551,24 @@ def foo_impl(x, y, z, w):
             getattr(torch.ops, self.test_ns).foo.default,
             getattr(torch.ops, self.test_ns).foo_functional.default, (x, y, z, w))
 
+    def test_register_fallthrough(self):
+        my_lib = Library('aten', 'IMPL')
+        my_lib.impl("mm", torch.library.Fallthrough(), "AutocastCPU")
+
```

Comment:
(no action required) I'm a bit surprised this works. The Dispatcher generally errors when someone tries to register two kernels for the same DispatchKey, but I guess it has some weird special-casing around fallthroughs.

Reply:
It warns but it doesn't raise right?

Reply:
Yea I think you can do this even for non-fallthroughs. For example

```
import torch
from torch.library import impl, Library

def dummy_func(a, b):
    return a + b

my_lib = Library('aten', 'IMPL')
my_lib.impl("mm", dummy_func, "CPU")

a = torch.arange(6, device='cpu', dtype=torch.float32).reshape(2, 3)
b = torch.arange(6, device='cpu', dtype=torch.float32).reshape(2, 3)

print(torch.mm(a, b))
```

Outputs

```
/data/users/mg1998/pytorch/torch/library.py:139: UserWarning: Warning only once for all operators,  other operators may also be overrided.
  Overriding a previously registered kernel for the same operator and the same dispatch key
  operator: aten::mm(Tensor self, Tensor mat2) -> Tensor
    registered at /data/users/mg1998/pytorch/build/aten/src/ATen/RegisterSchema.cpp:6
  dispatch key: CPU
  previous kernel: registered at /data/users/mg1998/pytorch/aten/src/ATen/LegacyBatchingRegistrations.cpp:1079
       new kernel: registered at /dev/null:53 (Triggered internally at /data/users/mg1998/pytorch/aten/src/ATen/core/dispatch/OperatorEntry.cpp:153.)
  self.m.impl(name, dispatch_key if dispatch_key != "" else "CompositeImplicitAutograd", fn)
tensor([[ 0.,  2.,  4.],
        [ 6.,  8., 10.]])
```

Reply:
Got it, thanks for the correction. The behavior makes sense to me then